meterfs/meterfs.c

```diff
@@ -1311,3 +1312,146 @@ int meterfs_init(meterfs_ctx_t *ctx)
 
 	return 0;
 }
+
+
+/* TODO Change meterfs interfaces to natively support libstorage. */
+/* Temporary adapters to implement storage fsops interfaces */
+
+
+static int adapter_open(void *info, oid_t *oid)
```

Comment:
@Maxez why doesn't these functions take `struct _storage_devCtx_t *info` instead of `void *info`?

Reply:
I don't think it would make sense in general case. This should be fs info not storage info. One can choose what content it holds via setting info fields of storage_fs_t struct (see line 1446). Eg jffs2 stores its own partition info there https://github.com/phoenix-rtos/phoenix-rtos-filesystems/blob/b699b05c6d8d7f97944771cdb57bb1659cd87696/jffs2/libjffs2.c#L1064.
However maybe adding new partial struct _storage_fs_info_t may be a good idea.

Reply:
This port of meterfs to be compatible with libstorage is kind of hacky. Ideally it should use mtd api if possible, should I look into porting it properly?